__tests__/feature/searchAndViewResource.test.js

```diff
@@ -0,0 +1,53 @@
+import { fireEvent, screen } from '@testing-library/react'
+import { renderApp } from 'testUtils'
+import * as sinopiaSearch from 'sinopiaSearch'
+import { resourceSearchResults } from 'fixtureLoaderHelper'
+import Config from 'Config'
+
+// This forces Sinopia server to use fixtures
+jest.spyOn(Config, 'useResourceTemplateFixtures', 'get').mockReturnValue(true)
+jest.mock('sinopiaSearch')
+// Mock jquery
+global.$ = jest.fn().mockReturnValue({ popover: jest.fn() })
+
+describe('searching and viewing a resource', () => {
+  sinopiaSearch.getTemplateSearchResults.mockResolvedValue({
+    totalHits: 0,
+    results: [],
+    error: undefined,
+  })
+
+  it('renders a modal without edit controls', async () => {
+    // Setup search component to return known resource
+    const uri = 'http://localhost:3000/resource/c7db5404-7d7d-40ac-b38e-c821d2c3ae3f'
+    sinopiaSearch.getSearchResultsWithFacets.mockResolvedValue(resourceSearchResults(uri))
+
+    renderApp()
+
+    fireEvent.click(screen.getByText('Linked Data Editor', { selector: 'a' }))
+    fireEvent.click(screen.getByText('Search', { selector: 'a' }))
+
+    fireEvent.change(screen.getByLabelText('Query'), { target: { value: uri } })
+    fireEvent.click(screen.getByTestId('Submit search'))
+
+    await screen.findByText(uri)
+    expect(screen.getByText('http://id.loc.gov/ontologies/bibframe/Fixture')).toBeInTheDocument()
+
+    // Modal hasn't rendered yet
+    expect(screen.queryByRole('dialog', { name: 'View Resource' })).not.toBeInTheDocument()
+    expect(screen.getByTestId('view-resource-modal').classList).not.toContain('show')
+
+    // Click the view icon next to the search result row
+    expect(screen.getByTestId(`View ${uri}`)).toBeInTheDocument()
+    fireEvent.click(screen.getByTestId(`View ${uri}`))
+
+    // Modal has now rendered
+    expect((await screen.findByTestId('view-resource-modal')).classList).toContain('show')
+    expect((await screen.findAllByText('Uber template1, property1', { selector: 'span' }))).toHaveLength(1)
+
```

Comment:
Should test that some of the form exists and is read only. (Just a sampling.)

Reply:
@justinlittman yeah, this is as far as I could get before the jest output became incomprehensible. I wanted to write a fuller test but was eager to stop beating my head against the test suite. I'll give it another go today.

This is also where I have found Cypress tests easier to work with than feature tests, because I don't have a good grasp on what's going on in these feature tests (what data is there? what does the page look like when the failure occurs? etc.). `screen.debug()` helps a bit, but then I'm scrolling through thousands of lines of HTML in my terminal trying to suss out what I'm looking at. Anyway, preaching to the choir here, and I reckon these are all familiar pain points!

Reply:
@justinlittman Done. Take another look?